src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -602,119 +543,104 @@ public class DosFileManager {
         return (uint)newOffset;
     }
 
-    private void SetOpenFile(ushort fileHandle, OpenFile? openFile) {
-        _openFiles[fileHandle] = openFile;
+    private void SetOpenFile(ushort fileHandle, OpenFile? openFile) => OpenFiles[fileHandle] = openFile;
+    
+    private void UpdateDosTransferAreaFromFile(string matchingFile) {
+        if (_loggerService.IsEnabled(Serilog.Events.LogEventLevel.Verbose)) {
+            _loggerService.Verbose("Found matching file {MatchingFile}", matchingFile);
+        }
+        DosDiskTransferArea dosDiskTransferArea = new(_memory, GetDiskTransferAreaAddressPhysical());
+        FileInfo attributes = new FileInfo(matchingFile);
+        DateTime creationZonedDateTime = attributes.CreationTimeUtc;
+        DateTime creationLocalDate = creationZonedDateTime.ToLocalTime();
+        DateTime creationLocalTime = creationZonedDateTime.ToLocalTime();
+        dosDiskTransferArea.FileDate = ToDosDate(creationLocalDate);
+        dosDiskTransferArea.FileTime = ToDosTime(creationLocalTime);
+        dosDiskTransferArea.FileSize = (ushort)attributes.Length;
+        dosDiskTransferArea.FileName = Path.GetFileName(matchingFile);
     }
 
-    private string? ToCaseSensitiveFileName(string? caseInsensitivePath) {
-        if (string.IsNullOrWhiteSpace(caseInsensitivePath)) {
-            return null;
+    /// <summary>
+    /// Creates a directory on disk.
+    /// </summary>
+    /// <param name="dosDirectory">The directory name to create</param>
+    /// <returns></returns>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult CreateDirectory(string dosDirectory) {
+        string? parentFolder = _dosPathResolver.TryGetFullHostParentPathFromDos(dosDirectory);
+        if (string.IsNullOrWhiteSpace(parentFolder)) {
+            return PathNotFoundError(dosDirectory);
         }
 
-        string fileToProcess = ConvertUtils.ToSlashPath(caseInsensitivePath);
-        string? parentDir = Path.GetDirectoryName(fileToProcess);
-        if (File.Exists(fileToProcess) || Directory.Exists(fileToProcess) ||
-            (string.IsNullOrWhiteSpace(parentDir) == false && Directory.Exists(parentDir) && Directory.GetDirectories(parentDir).Length == 0)) {
-            // file exists or root reached, no need to go further. Path found.
-            return caseInsensitivePath;
+        if (!Directory.Exists(parentFolder)) {
+            CreateDirectory(_dosPathResolver.GetRelativeHostPathToCurrentDirectory(parentFolder));
```

Comment:
Why do we need to recurse? Wouldn't `Directory.CreateDirectory(parentFolder)` work here?

Reply:
I don't think it's accepted on all platforms. On Windows, sure. But it may not be so on Unix.

Reply:
I don't see why not. I would assume the same restrictions apply to `Directory.Exists(parentFolder)`.

Reply:
Recursion removed.